### PR TITLE
Lac rm hapi

### PR DESCRIPTION
**Pull request by kinghuin**
### PR types
Function optimization

### PR changes
Models

### Describe
1. lac的数据集管理采用最新版的数据集管理方式
2. lac不再使用hapi，crf、chunk evaluator去掉冗余参数

### CODE REVIEW

File: examples/lexical_analysis/data.py [hunk 1]
```diff
@@ -17,135 +17,107 @@
...
+    if labels is not None:
+        labels = labels[:max_seq_len]
+        label_ids = convert_tokens_to_ids(
+            labels, label_vocab, oov_replace_token="O")
+        return token_ids, length, label_ids
+    else:
+        return token_ids, length
 
 
 def parse_lac_result(words, preds, lengths, word_vocab, label_vocab):
```
Comment (ZeyuChen) [inline examples/lexical_analysis/data.py]: 是否去掉lac这些术语？用更明了的语句来命名函数

File: examples/lexical_analysis/predict.py [hunk 2]
```diff
@@ -15,14 +15,15 @@
...
+from functools import partial
 
 import numpy as np
 import paddle
 from paddle.static import InputSpec
 from paddlenlp.data import Pad, Tuple, Stack
 from paddlenlp.metrics import ChunkEvaluator
 
-from data import LacDataset, parse_lac_result
+from data import load_dataset, load_vocab, convert_example, parse_lac_result
```
Comment (ZeyuChen) [inline examples/lexical_analysis/predict.py]: `parse_lac_result` -> `parse_result`

File: paddlenlp/metrics/chunk.py [hunk 3]
```diff
@@ -45,7 +46,14 @@ def __init__(self, label_list, suffix=False):
         self.num_label_chunks = 0
         self.num_correct_chunks = 0
 
-    def compute(self, inputs, lengths, predictions, labels):
+    def compute(self, lengths, predictions, labels, redundant=None):
+        if redundant is not None:
+            redundant, lengths, predictions, labels = lengths, predictions, labels, redundant
+            if not getattr(self, "has_warn", False):
+                logger.warning(
+                    'Compatibility Warning: The params of ChunkEvaluator.compute has been modified. The old version is inputs, lengths, predictions, labels while the current version is lengths, predictions, labels.  Please update the usage.'
```
Comment (ZeyuChen) [inline paddlenlp/metrics/chunk.py]: 对于变量加一下  ``括起来比较清晰

**Review by ZeyuChen — COMMENTED** [review]
修改一下一些基础函数命名。

File: examples/lexical_analysis/data.py (hunk 1 above)
Comment (kinghuin) [reply examples/lexical_analysis/data.py]: DONE

File: examples/lexical_analysis/predict.py (hunk 2 above)
Comment (kinghuin) [reply examples/lexical_analysis/predict.py]: DONE

File: paddlenlp/metrics/chunk.py (hunk 3 above)
Comment (kinghuin) [reply paddlenlp/metrics/chunk.py]: DONE

File: examples/lexical_analysis/eval.py [hunk 4]
```diff
@@ -43,7 +44,19 @@ def evaluate(args):
     paddle.set_device(args.device)
 
     # create dataset.
-    test_dataset = LacDataset(args.data_dir, mode='test')
+    test_ds = load_dataset(datafiles=(os.path.join(args.data_dir, 'test.tsv')))
+    word_vocab = load_vocab(os.path.join(args.data_dir, 'word.dic'))
+    label_vocab = load_vocab(os.path.join(args.data_dir, 'tag.dic'))
+    token_replace_vocab = load_vocab(os.path.join(args.data_dir, 'q2b.dic'))
```
Comment (ZeyuChen) [inline examples/lexical_analysis/eval.py]: 对这个文件加载添加下注释，因为q2b有点tricky。
全角 SBC case 
半角 DBC case

Comment (ZeyuChen) [inline examples/lexical_analysis/eval.py]: token_replace_vocab -> normalize_vocab

File: paddlenlp/metrics/chunk.py [hunk 5]
```diff
@@ -45,7 +46,14 @@ def __init__(self, label_list, suffix=False):
         self.num_label_chunks = 0
         self.num_correct_chunks = 0
 
-    def compute(self, inputs, lengths, predictions, labels):
+    def compute(self, lengths, predictions, labels, redundant=None):
+        if redundant is not None:
```
Comment (ZeyuChen) [inline paddlenlp/metrics/chunk.py]: 加一下对于Compatibility的注释。
例如之前的注释顺序是什么，`redundant`->`dummy`

File: examples/lexical_analysis/train.py [hunk 6]
```diff
@@ -60,58 +92,63 @@ def train(args):
...
+    last_step = args.epochs * len(train_loader)
+    tic_train = time.time()
+    for epoch in range(args.epochs):
+        for step, batch in enumerate(train_loader):
+            global_step += 1
+            token_ids, length, label_ids = batch
+            loss = model(token_ids, length, label_ids)
+            avg_loss = paddle.mean(loss)
+            if global_step % args.logging_steps == 0:
+                print("global step %d / %d, loss: %f, speed: %.2f step/s" %
```
Comment (ZeyuChen) [inline examples/lexical_analysis/train.py:141]: LOG保持跟benchmark一致吧，但是hapi应该是跟benchmark一致的了。

File: examples/lexical_analysis/data.py [hunk 7]
```diff
@@ -17,138 +17,115 @@
...
+            elif len(terms) == 1:
+                key, value = terms[0], i
+            else:
+                raise ValueError("Error line: %s in file: %s" %
+                                 (line, dict_path))
+            vocab[key] = value
+    return vocab
+
+
+def normalize_token(token, token_replace_vocab):
```
Comment (ZeyuChen) [inline examples/lexical_analysis/data.py]: 添加注释
`# Normalize text from SBC case  to DBC case`

File: examples/lexical_analysis/README.md [hunk 8]
```diff
@@ -84,7 +84,7 @@ python -m paddle.distributed.launch --gpus "0,1"  train.py \
 
 ```bash
 python eval.py --data_dir ./lexical_analysis_dataset_tiny \
-        --init_checkpoint ./save_dir/final \
+        --init_checkpoint ./save_dir/model_100.pdparams \
```
Comment (wawltor) [inline examples/lexical_analysis/README.md:87]: 这总决定数字的path路径，最好能说明一下，用户经过训练之后才能获取到模型

File: examples/lexical_analysis/data.py (hunk 7 above)
Comment (ZeyuChen) [reply examples/lexical_analysis/data.py]: token_replace_vocab改掉。理论上应该叫normlize_vocab，例如后续繁简统一等扩展都是可以通过词表映射方式完成

File: examples/lexical_analysis/train.py [hunk 9]
```diff
@@ -60,58 +92,63 @@ def train(args):
...
+                      (global_step, last_step, avg_loss,
+                       args.logging_steps / (time.time() - tic_train)))
+                tic_train = time.time()
+            avg_loss.backward()
+            optimizer.step()
+            optimizer.clear_grad()
+            if global_step % args.save_steps == 0 or global_step == last_step:
+                if paddle.distributed.get_rank() == 0:
+                    evaluate(model, chunk_evaluator, test_loader)
```
Comment (wawltor) [inline examples/lexical_analysis/train.py:150]: 这块要用一下no_grad修饰一下

File: paddlenlp/metrics/chunk.py [hunk 10]
```diff
@@ -45,7 +46,14 @@ def __init__(self, label_list, suffix=False):
...
+            redundant, lengths, predictions, labels = lengths, predictions, labels, redundant
+            if not getattr(self, "has_warn", False):
+                logger.warning(
+                    'Compatibility Warning: The params of ChunkEvaluator.compute has been modified. The old version is `inputs`, `lengths`, `predictions`, `labels` while the current version is `lengths`, `predictions`, `labels`.  Please update the usage.'
+                )
+                self.has_warn = True
```
Comment (wawltor) [inline paddlenlp/metrics/chunk.py:70]: 这块的主要修改的目的是什么，没有太看懂这块的逻辑

File: examples/lexical_analysis/eval.py (hunk 4 above)
Comment (kinghuin) [reply examples/lexical_analysis/eval.py]: 反了  https://en.wikipedia.org/wiki/DBCS#In_CJK_(Chinese/Japanese/Korean)_computing
"A DBCS supports national languages that contain many unique characters or symbols (the maximum number of characters that can be represented with one byte is 256 characters, while two bytes can represent up to 65,536 characters). Examples of such languages include Japanese and Chinese."

Comment (kinghuin) [reply examples/lexical_analysis/eval.py]: DONE

File: paddlenlp/metrics/chunk.py (hunk 5 above)
Comment (kinghuin) [reply paddlenlp/metrics/chunk.py]: DONE

File: examples/lexical_analysis/train.py (hunk 6 above)
Comment (kinghuin) [reply examples/lexical_analysis/train.py:141]: 在benchmark的基础上去掉了epoch和batch，个人感觉加了特别难看

File: examples/lexical_analysis/data.py (hunk 7 above)
Comment (kinghuin) [reply examples/lexical_analysis/data.py]: DONE

File: examples/lexical_analysis/README.md (hunk 8 above)
Comment (kinghuin) [reply examples/lexical_analysis/README.md:87]: DONE

File: examples/lexical_analysis/train.py (hunk 9 above)
Comment (kinghuin) [reply examples/lexical_analysis/train.py:150]: DONE

File: paddlenlp/metrics/chunk.py (hunk 10 above)
Comment (kinghuin) [reply paddlenlp/metrics/chunk.py:70]: 以前为了兼容hapi写法，这个函数有一个冗余参数`inputs`，这次修改是为了去掉冗余参数，同时兼容老版本的冗余参数的写法

**Review by ZeyuChen — APPROVED** [review]
LGTM
